Euphorolog.Services/Services/UsersService.cs

```diff
@@ -13,36 +20,129 @@ public class UsersService : IUsersService
     {
         public readonly EuphorologContext _context;
         public readonly IUsersRepository _usersRepository;
-        public UsersService(EuphorologContext context, IUsersRepository usersRepository)
+        public readonly IConfiguration _configuration;
+        public readonly IHttpContextAccessor _httpContextAccessor;
+        public readonly IMapper _mapper;
+        public UsersService(
+            EuphorologContext context,
+            IUsersRepository usersRepository,
+            IConfiguration configuration,
+            IHttpContextAccessor httpContextAccessor,
+            IMapper mapper)
         {
             _context = context;
             _usersRepository = usersRepository;
+            _configuration = configuration;
+            _httpContextAccessor = httpContextAccessor;
+            _mapper = mapper;
         }
-        public async Task<List<Users>> GetAllUsersAsync()
+        public async Task<List<UserInfoResponseDTO>> GetAllUsersAsync()
         {
             var ret = await _usersRepository.GetAllUsersAsync();
-            return ret;
+            return _mapper.Map<List<UserInfoResponseDTO>>(ret);
 
         }
-        public async Task<Users> GetUserByIdAsync(string id)
+        public async Task<UserInfoResponseDTO> GetUserByIdAsync(string id)
         {
             var ret = await _usersRepository.GetUserByIdAsync(id);
-            return ret;
+            if(ret == null)
+            {
+                throw new NotFoundException("User doesn't Exist");
+            }
+            return _mapper.Map<UserInfoResponseDTO>(ret);
 
         }
-        public async Task<List<Users>> CreateUserAsync(Users user)
+        public async Task DeleteUserAsync(string id)
         {
-            user.passChangedAt = DateTime.UtcNow;
-            user.passChangedflag = true;
-            return await _usersRepository.CreateUserAsync(user);
-        }
-        public async Task<List<Users>> DeleteUserAsync(string id)
-        {
-            return await _usersRepository.DeleteUserAsync(id);
+            /*
+             * If the user is logged in
+             * If the token is valid based on time
+             * If the user Exists
+             * If the id is their's
+             */
+
+            // User logged In
+            var username = _httpContextAccessor.HttpContext?.User?.FindFirst(ClaimTypes.Name)?.Value;
+            if(username == null)
+            {
+                throw new UnAuthorizedException("Log In First!");
```

Comment:
Service methods should not check if the user is logged in. You have done it in the controller already, with `Authorized` attribute.